src/collections/blog/2022/12-23/index.md

```diff
@@ -0,0 +1,114 @@
+
+The Common Expression Language (CEL) is a policy language developed by Google that is used for evaluating expressions in a variety of contexts. CEL is designed to be simple, flexible, and efficient, with a focus on providing a consistent way to evaluate expressions across different platforms and environments.
+
+One key feature of CEL is its support for custom functions, which allows users to define their own functions and use them in expressions. This enables users to extend the functionality of CEL and tailor it to their specific needs.
+
+HashiCorp Sentinel is another policy language that is commonly used for evaluating expressions in various contexts. Sentinel is designed to be flexible and extensible, with a focus on providing a consistent way to define and enforce policies across different platforms and environments.
+
+Like CEL, Sentinel supports custom functions and allows users to define their own functions and use them in expressions. In addition, Sentinel provides support for a wide range of operators and data types, making it a powerful tool for evaluating complex expressions.
+
+Open Policy Agent Rego is another policy language that is widely used for evaluating expressions and enforcing policies. Rego is designed to be simple and easy to use, with a focus on providing a consistent way to define and enforce policies across different platforms and environments.
+
+Like CEL and Sentinel, Rego supports custom functions and allows users to define their own functions and use them in expressions. In addition, Rego provides support for a wide range of operators and data types, making it a powerful tool for evaluating complex expressions.
+
+## Code Comparisons
+
+To give an example of how these policy languages can be used, consider a scenario in which we want to enforce a policy that allows users to access a certain resource only if they are authorized to do so.
+
+In CEL, we might define the policy as follows:
+
+```function isAuthorized(user: User) -> bool {
+  return user.role == "admin" || user.role == "moderator";
+}
+
+expression authorizedToAccessResource(user: User, resource: Resource) -> bool {
+  return isAuthorized(user) && user.hasPermission(resource);
+}```
+
+In Sentinel, we might define the policy as follows:
+
+```import "strings"
+
+authorization_roles = ["admin", "moderator"]
+
+is_authorized(user) {
+  user.role in authorization_roles
+}
+
+authorized_to_access_resource(user, resource) {
+  is_authorized(user) && user.has_permission(resource)
+}```
+
+In Rego, we might define the policy as follows:
+
+```
+package policy
+
+import data.users
+
+authorization_roles = ["admin", "moderator"]
+
+is_authorized(user) {
+  user.role in authorization_roles
+}
+
+authorized_to_access_resource(user, resource) {
+  is_authorized(user) && user.has_permission(resource)
+}
+```
+
```

Comment:
```suggestion

```
function isAuthorized(user: User) -> bool {
  return user.role == "admin" || user.role == "moderator";
}

expression authorizedToAccessResource(user: User, resource: Resource) -> bool {
  return isAuthorized(user) && user.hasPermission(resource);
}
```

In Sentinel, we might define the policy as follows:

```
import "strings"

authorization_roles = ["admin", "moderator"]

is_authorized(user) {
  user.role in authorization_roles
}

authorized_to_access_resource(user, resource) {
  is_authorized(user) && user.has_permission(resource)
}
```

In Rego, we might define the policy as follows:

```
package policy

import data.users

authorization_roles = ["admin", "moderator"]

is_authorized(user) {
  user.role in authorization_roles
}

authorized_to_access_resource(user, resource) {
  is_authorized(user) && user.has_permission(resource)
}
```

```